src/coffea/dataset_tools/preprocess.py

```diff
@@ -314,7 +314,20 @@ def preprocess(
         dak_norm_files = dask_awkward.from_awkward(
             ak_norm_files, math.ceil(len(ak_norm_files) / files_per_batch)
         )
-        files_to_preprocess[name] = dask_awkward.map_partitions(
+
+        concat_fn = partial(
+            awkward.concatenate,
+            axis=0,
+        )
+
+        split_every = 8
+
+        files_trl_label = f"{name}"
+        files_trl_token = dask.base.tokenize(dak_norm_files, concat_fn, split_every)
+        files_trl_name = f"{files_trl_label}-{files_trl_token}"
+        files_trl_tree_node_name = f"{files_trl_label}-tree-node-{files_trl_token}"
```

Comment:
These use essentially the fileset keys to identify the names, which looked fine to me in the task graph but I'm happy for these to change as desired.